raiden-ts/src/utils/rx.ts

```diff
@@ -66,23 +68,32 @@ export function distinctRecordValues<R>(
 /**
  * Operator to repeat-subscribe an input observable until a notifier emits
  *
- * @param notifier - Notifier observable
- * @param delayMs - Delay between retries
- * @returns Monotype operator
+ * @param notifier - Notifier observable to stop repeating
+ * @param delayMs - Delay between retries; in milliseconds, or an Iterator of delays
+ * @returns Monotype operator function
  */
 export function repeatUntil<T>(
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  notifier: Observable<any>,
-  delayMs = 30e3,
+  notifier: Observable<unknown>,
+  delayMs: number | Iterator<number> = 30e3,
 ): MonoTypeOperatorFunction<T> {
-  // Resubscribe/retry every 30s after messageSend succeeds
+  // Resubscribe/retry every 30s or yielded ms after messageSend succeeds
   // Notice first (or any) messageSend.request can wait for a long time before succeeding, as it
   // waits for address's user in transport to be online and joined room before actually
   // sending the message. That's why repeatWhen emits/resubscribe only some time after
   // sendOnceAndWaitSent$ completes, instead of a plain 'interval'
   return (input$) =>
     input$.pipe(
-      repeatWhen((completed$) => completed$.pipe(delay(delayMs))),
+      repeatWhen((completed$) =>
+        completed$.pipe(
+          map(() => {
+            if (typeof delayMs === 'number') return delayMs;
+            const next = delayMs.next();
+            return !next.done ? next.value : -1;
+          }),
+          takeWhile((value) => value >= 0), // stop repeatWhen when done
```

Comment:
When will this become true for the plain number? :thinking:

Reply:
So in this case it would depend on the `notifier`?

Reply:
It already does most of the time, even if exponential backoff. This `takeWhile` is just a way to be more consistent with the introduced generator, but usually the one to tell to stop is `notifier`, e.g.: a `Processed` message came for that LockedTransfer, give up retrying because we're good already.